### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by masteryada378**
First PR

### CODE REVIEW

File: source/js/_module.js [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    for (let i = 0; i <arrayData.length; i++){
+        arrayData[i].idIndex = i;
+        arrayData[i].inCart = false;
+    }
+} 
+addIdNumberAndFlag(phonesArr);
+let newPhones = [];
+let renderItems = (arrayPhone) => {
+    
+    template = ['','','','','','']; // ??????? 2 дня потратил
```
Comment (danya-prohvat) [inline source/js/_module.js]: dont write comment especially in russian

File: source/js/_module.js [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    cartClose: document.getElementById('cart__close'),
+    cartWrapper: document.getElementById('cart__wrapper'),
+    cartDashboard: document.getElementById('dashboard'),
+    cartContent: document.getElementById('cart__content'),
+    bodyElement: document.body
+}
+let cartElementString = '';
+let cartTemplate = [];
+let template = ['','','','','','']; // если так не делать будет undefinde
+let phonesArr = [
```
Comment (danya-prohvat) [inline source/js/_module.js:17]: if you wont change data use const

File: source/js/_module.js [hunk 3]
```diff
@@ -0,0 +1,238 @@
+"use strict";
+console.log('Hello friend');
```
Comment (danya-prohvat) [inline source/js/_module.js]: remove console.log when push commits pls

File: source/js/_module.js [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+let elements = {
```
Comment (danya-prohvat) [inline source/js/_module.js]: const

File: source/js/_module.js [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    cartBtn: document.getElementById('header-cart'),
+    cartClose: document.getElementById('cart__close'),
+    cartWrapper: document.getElementById('cart__wrapper'),
+    cartDashboard: document.getElementById('dashboard'),
+    cartContent: document.getElementById('cart__content'),
+    bodyElement: document.body
+}
+let cartElementString = '';
+let cartTemplate = [];
+let template = ['','','','','','']; // если так не делать будет undefinde
```
Comment (danya-prohvat) [inline source/js/_module.js]: comment

File: source/js/_module.js [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+    {"name": "Microsoft Surface Laptop 3 - 13.5\"", "price": "$1260.00", "lowerPrice": "", "photo": "note1.jpg"},
+    {"name": "Microsoft Surface Book 3", "price": "$1150.00", "lowerPrice": "", "photo": "note2.jpg"},
+    {"name": "Microsoft Surface Pro 7 Platinum", "price": "$987.00", "lowerPrice": "", "photo": "note3.jpg"},
+    {"name": "Microsoft Surface Laptop GO", "price": "$1320.00", "lowerPrice": "", "photo": "note4.jpg"},
+    {"name": "Xiaomi Mi RedmiBook 14\"", "price": "$1230.00", "lowerPrice": "", "photo": "note5.jpg"},
+    {"name": "Xiaomi Mi Notebook Pro 15.6", "price": "$1110.00", "lowerPrice": "", "photo": "note6.jpg"},
+    {"name": "Xiaomi Mi RedmiBook 16\"", "price": "$1140.00", "lowerPrice": "", "photo": "note7.jpg"},
+    {"name": "Xiaomi Mi RedmiBook 13\"", "price": "$1340.00", "lowerPrice": "", "photo": "note8.jpg"}
+]
+let addIdNumberAndFlag = (arrayData)=>{
```
Comment (danya-prohvat) [inline source/js/_module.js]: in function expression use const pls

File: source/js/_module.js [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+} 
+addIdNumberAndFlag(phonesArr);
+let newPhones = [];
+let renderItems = (arrayPhone) => {
+    
+    template = ['','','','','','']; // ??????? 2 дня потратил 
+
+    let newIndex = -1;
+    for (let i = 0; i < arrayPhone.length; i++) {
+        if((i % 8) == 0) {
```
Comment (danya-prohvat) [inline source/js/_module.js]: you can remove {} when single-line expression after conditional

File: source/js/_module.js [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+        }
+        pushCart();
+        elements.cartContent.innerHTML = cartElementString;
+        elements.phoneBox.innerHTML = '';
+        elements.pagination.innerHTML = '';
+        initList(phonesArr);
+    }
+}
+let deleteFromCart = (id)=>{
+    // console.log(phonesArr[id]);
```
Comment (danya-prohvat) [inline source/js/_module.js:206]: dont push commented code in review

File: source/js/_module.js [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+    }
+};
+initList(phonesArr);
+let searchHandler = (event)=> {
+
+    newPhones = [];
+    let searchSimvol = event.target.value.toLowerCase();
+    let dataInputStr = ''; 
+
+    for (let i=0; i < phonesArr.length; i++) {
```
Comment (danya-prohvat) [inline source/js/_module.js]: take look at forEach

File: source/js/_module.js [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+                        </div>`;
+};
+let makePaginator = (phonesListLenghtArg) => {
+    for (let i = 0; i < phonesListLenghtArg; i++) {
+        elements.pagination.innerHTML += `<button>${i+1}</button>`;
+    }
+    if(!!elements.pagination.children[0]){
+        elements.pagination.children[0].classList.add('active');
+    } else {
+        console.log('errorList');
```
Comment (danya-prohvat) [inline source/js/_module.js]: also console.log

File: source/js/_module.js (hunk 1 above)
Comment (masteryada378) [reply source/js/_module.js]: done

File: source/js/_module.js (hunk 2 above)
Comment (masteryada378) [reply source/js/_module.js:17]: The data is changed in the addIdNumberAndFlag function, I add "idIndex" and "inCart"

File: source/js/_module.js (hunk 3 above)
Comment (masteryada378) [reply source/js/_module.js]: done

File: source/js/_module.js (hunk 4 above)
Comment (masteryada378) [reply source/js/_module.js]: done

File: source/js/_module.js (hunk 5 above)
Comment (masteryada378) [reply source/js/_module.js]: done

File: source/js/_module.js (hunk 6 above)
Comment (masteryada378) [reply source/js/_module.js]: done

File: source/js/_module.js (hunk 7 above)
Comment (masteryada378) [reply source/js/_module.js]: done

File: source/js/_module.js (hunk 8 above)
Comment (masteryada378) [reply source/js/_module.js:206]: done

File: source/js/_module.js (hunk 10 above)
Comment (masteryada378) [reply source/js/_module.js]: done

File: source/js/_module.js (hunk 9 above)
Comment (masteryada378) [reply source/js/_module.js]: done

**Pull request merged**